### PR TITLE
Prepare for OAuth2 flow

### PR DESCRIPTION
**Pull request by tokee**
The Ds-services are expected to be OAuth2 enabled in the future. This is needed for some endpoints, where special privileges should disable licence checks for exports; needed for the initial update of the Solr index as the Solr index is used by the license module - chicken & egg.

This pull request adds support for the HTTP header
```
Simulated-OAuth2-Group: internal_service
```
when calling web service endpoints. If the header is set, the license check is bypassed. This is **absolutely not** an attempt at providing access checks, but serves to test the flow of headers between our services as well as having different groups of callers.

For debugging, the endpoint `/monitor/whoami` has been added. It can be tested with 
```
curl -X GET "http://localhost:9073/ds-present/v1/probe/whoami" -H  "Simulated-OAuth2-Group: foo"
```

### CODE REVIEW

File: src/main/java/dk/kb/present/util/DsPresentClient.java [hunk 1]
```diff
@@ -58,40 +63,75 @@ public class DsPresentClient extends DsPresentApi {
...
+
+    /**
+     * Create a client for the remote ds-present service.
      * <p>
-     * When working with YAML configs, it is suggested to define the ds-present URI as the structure
+     * This constructor expects the structure
      * <pre>
-     * storage:
-     *   url: 'http://localhost:9073/ds-storage/v1'
+     * config:
```
Comment (VictorHarbo) [inline src/main/java/dk/kb/present/util/DsPresentClient.java:75]: Config shouldn't contain the config: key

File: src/main/java/dk/kb/present/util/DsPresentClient.java [hunk 2]
```diff
@@ -58,40 +63,75 @@ public class DsPresentClient extends DsPresentApi {
...
-                setScheme(serviceURI.getScheme()).
-                setHost(serviceURI.getHost()).
-                setPort(serviceURI.getPort()).
-                setBasePath(serviceURI.getRawPath());
+    /**
+     * Create a client for the remote ds-present service.
+     * <p>
+     * When working with YAML configs, it is suggested to use the structure
+     * <pre>
+     * config:
```
Comment (VictorHarbo) [inline src/main/java/dk/kb/present/util/DsPresentClient.java:110]: Same comment as above

File: src/main/java/dk/kb/present/webservice/AccessUtil.java [hunk 3]
```diff
@@ -35,6 +36,10 @@
 public class AccessUtil {
     private static final Logger log = LoggerFactory.getLogger(AccessUtil.class);
 
+    public static final String HEADER_SIMULATED_GROUP = "Simulated-OAuth2-Group";
+    public static final String GROUP_INTERNAL_SERVICE = "internal_service";
+    public static final String GROUP_ADMIN = "admin";
```
Comment (VictorHarbo) [inline src/main/java/dk/kb/present/webservice/AccessUtil.java:41]: Question: How have these values been chosen?

File: src/main/openapi/ds-present-openapi_v1.yaml [hunk 4]
```diff
@@ -377,6 +383,20 @@ paths:
...
+        - Service
+      summary: 'Pseudo-OAuth2 check: The groups that the caller belongs'
+      operationId: whoami
+      responses:
+        '200':
+          description: 'OK'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Whoami'
```
Comment (VictorHarbo) [inline src/main/openapi/ds-present-openapi_v1.yaml:399]: I like this endpoint, works like a charm

**Review by VictorHarbo — CHANGES_REQUESTED** [review]
Code looks good. Minor comments. Also, a test fails after updating aegis and running kb init:  DsPresentClientTest.testRemoteRecordsStream:88 The requested number of records should be received ==> expected: <3> but was: <0>

File: src/main/java/dk/kb/present/webservice/AccessUtil.java (hunk 3 above)
Comment (tokee) [reply src/main/java/dk/kb/present/webservice/AccessUtil.java:41]: By careful deliberation and pondering for about 5 seconds.

They will be replaced later on. I'll add a comment about it to the code.

**Review by VictorHarbo — APPROVED** [review]
LGTM!

**Pull request merged**